### PR TITLE
Re #560: --redo-import now rewrites the index and prunes side chains

### PR DESCRIPTION
**Pull request by AlexandraRoatis**
## Description

- Corrected the bug where multiple index entries were introduced for the same block during the rebuild.
- Also rewriting the `index` database and removing side chain blocks from the `block` database.

## Type of change

Insert **x** into the following checkboxes to confirm (eg. [x]):
- [x] Bug fix.
- [ ] New feature.
- [x] Enhancement.
- [ ] Unit test.
- [ ] Breaking change (a fix or feature that causes existing functionality to not work as expected).
- [ ] Requires documentation update.

## Testing

- Ran the command on a correct database with side chains and one with incorrect index entries. Ensured that both databases are correct after the `--redo-import` execution with start from the genesis block and start from a block > 0.

### CODE REVIEW

File: modAionImpl/src/org/aion/zero/impl/db/AionBlockStore.java [hunk 1]
```diff
@@ -1250,6 +1251,36 @@ public void correctMainChain(AionBlock block, Logger log) {
...
+
+                // delete all the side-chain blocks
+                for (BlockInfo blockInfo : level) {
+                    if (!Arrays.equals(currentHash, blockInfo.getHash())) {
+                        blocks.delete(blockInfo.getHash());
+                    }
+                }
+
+                // replace all the block info with empty list
+                index.set(block.getNumber(), Collections.emptyList());
```
Comment (AionJayT) [inline modAionImpl/src/org/aion/zero/impl/db/AionBlockStore.java]: Looks like it only reset the index db at the reimport point level, where is the logic to reset the level between the reimport point to the current index DB height?

Comment (AlexandraRoatis) [reply modAionImpl/src/org/aion/zero/impl/db/AionBlockStore.java]: before the block import, the index is cleared and side-chains are pruned, then the block is imported which sets the correct index value

Comment (AlexandraRoatis) [reply modAionImpl/src/org/aion/zero/impl/db/AionBlockStore.java]: the index is corrected for each block that is imported, it does not modify the database for blocks predating the first block to re-import

Comment (AionJayT) [reply modAionImpl/src/org/aion/zero/impl/db/AionBlockStore.java]: Okay. understood.

**Pull request merged**